#162234049 Add recently published articles section

ucheg6 opened the pull request:
#### What does this PR do?
Add recently published articles section
#### Description of Task to be completed?
- create and render recent Article component
- create reducer and action types
- create recent articles saga and dispatch action to fetch recently published articles

#### How should this be manually tested?
On command line ;
- git clone this repo
- npm install 
- Run `npm start:dev ` from the `haven-ah-frontend` directory

#### What are the relevant pivotal tracker stories?
[#162234049](https://www.pivotaltracker.com/story/show/1162234049)

#### Screenshots (if appropriate)>
<img width="1323" alt="screen shot 2018-12-03 at 4 56 28 pm" src="https://user-images.githubusercontent.com/34472497/49384945-75a9b000-f71c-11e8-8091-a46f928e763d.png">

wiztemple commented:
Good Job @ucheg6 .

seyi-adeleke commented on src/components/containers/recentArticles/RecentArticles.jsx:
```diff
@@ -0,0 +1,57 @@
+          < div className = "columns is-variable is-8 mt-3">
+            {
+              this.props.articles
+            && this.props.articles.slice(0, 3).map((article, index) => {
```
Good job using Slice here. However you shouldn't  do this in the component. Either at the reducer or the store. Also shouldn't the api be returning the exact number of articles you need?

ucheg6 replied:
The get all articles api endpoint returns more than what I need. So I'm using the slice to display what I need

ucheg6 replied:
Yeah, the API returns the exact number of articles we need. The slice is just for displaying the cards. The Bulma style is expecting three cards in one row. If I don't control the display it displays 6 cards in one row.

seyi-adeleke commented on src/components/containers/recentArticles/RecentArticles.jsx:
```diff
@@ -0,0 +1,57 @@
+          < div className = "columns is-variable is-8 mt-3">
+            {
+              this.props.articles
+            && this.props.articles.slice(0, 3).map((article, index) => {
```
Also its not a good idea to use the index as a key https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

ucheg6 replied:
Thanks, I've changed it to articleId.

seyi-adeleke commented on src/components/containers/recentArticles/RecentArticles.jsx:
```diff
@@ -0,0 +1,57 @@
+          < div className = "columns is-variable is-8 mt-3">
+            {
+              this.props.articles
+            && this.props.articles.slice(3, 6).map((article, index) => {
```
🤔

ucheg6 replied:
So I am doing the same thing here. Rendering the last three on a different row.

seyi-adeleke commented on src/components/containers/recentArticles/recentArticles.css:
```diff
@@ -0,0 +1,68 @@
+body {
+    padding: 0;
+    margin: 0;
+    /* font-family: 'Playfair Display', sans-serif; */
```
shouldn't the font be the same?

seyi-adeleke replied:
Also your package lock shouldn't have a diff

ucheg6 replied:
The fonts are the same. the font is coming from APP.scss.

seyi-adeleke commented on src/App.jsx:
```diff
@@ -3,10 +3,9 @@ import { BrowserRouter as Router } from 'react-router-dom';
-
 import '../public/styles/App.scss';
 import '../public/styles/responsive.scss';
+import 'font-awesome/css/font-awesome.min.css';
```
Same thing about absolute and relative imports

seyi-adeleke commented on src/components/containers/recentArticles/RecentArticles.jsx:
```diff
@@ -0,0 +1,57 @@
+
+RecentArticles.propTypes = {
+  recentArticlesAction: PropTypes.func.isRequired,
+  articles: PropTypes.array,
```
add `.isRequired`

seyi-adeleke commented on src/components/containers/recentArticles/recentArticlesCard.jsx:
```diff
@@ -0,0 +1,45 @@
+};
+
+recentArticlesCard.propTypes = {
+  article: PropTypes.object,
```
add proptypes for the other props you're using

TheoOkafor reviewed (commented):
looks good.

ucheg6 committed:
feat(recent-articles): Add recently published articles section

create and render recent Article component
create reducer and action types
create recent articles saga and dispatch action to fetch recently published articles
